### ISSUE TITLE
CRD generation should Types with fields with custom marshaller/unmarshaller 

### ISSUE THREAD

**Issue by droot** [kubernetes-sigs/kubebuilder#406]
If custom type uses fields with custom marshaller/unmarshaller such as `resource.Quantity` or `metav1.Duration`, the validation schema for the generated CRD will have `type: "object"`. Submitting a CR instance with string representation of those fields such as "10G" or "11s", throws an error 
```
: validation failure list:
spec.timeTest in body must be of type object: "string"
```

Type definition:

```Go
import (
    "k8s.io/apimachinery/pkg/api/resource"
    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

type MySQLSpec struct {
    DiskSize resource.Quantity `json:"diskSize,omitempty"`
    TimeTest metav1.Duration `json:"timeTest,omitempty"`
}
```

Task is figure out a solution for this use-case.

**Comment by droot** [comment]
Current workaround:

Represent the field as `string` type and use the parse method such as `ParseDuration`, `ParseQuantity` to convert to a type and use it the implementation. Use `Pattern` annotation to validate the input.

**Comment by tamalsaha** [comment]
Kubebuilder now generates valid oepnapi schema for both of these types.

- https://github.com/kubernetes-sigs/controller-tools/pull/174
- https://github.com/kubernetes-sigs/controller-tools/pull/138

**Comment by tamalsaha** [comment]
As @munnerz mentions in https://github.com/kubernetes-sigs/kubebuilder/issues/645 , open api type can be supported via a `kubebuilder:type` tag.

**Comment by DirectXMan12** [comment]
closing as per the above comments.

**Comment by DirectXMan12** [comment]
keeping open for generate type info tag.

**Comment by DirectXMan12** [comment]
closing, fixed as of v0.2.0-xyz
